pam/integration-tests/vhs-helpers_test.go

```diff
@@ -178,6 +179,29 @@ func (td tapeData) RunVhs(t *testing.T, testType vhsTestType, outDir string, cli
 
 	cmd.Args = append(cmd.Args, td.PrepareTape(t, testType, outDir))
 	out, err := cmd.CombinedOutput()
+
+	isSSHError := func(processOut []byte) bool {
+		const sshConnectionResetByPeer = "Connection reset by peer"
+		const sshConnectionClosed = "Connection closed by"
+		output := string(processOut)
+		if strings.Contains(output, sshConnectionResetByPeer) {
+			return true
+		}
+		return strings.Contains(output, sshConnectionClosed)
+	}
+	if err != nil && testType == vhsTestTypeSSH && isSSHError(out) {
+		t.Logf("SSH Connection failed on tape %q: %v: %s", td.Name, err, out)
+		// Let's repeat first, if it fails again, let's just skip the test.
+		//nolint:gosec // G204 it's a test and we explicitly set the parameters before.
+		newCmd := exec.Command(cmd.Args[0], cmd.Args[1:]...)
+		newCmd.Dir = cmd.Dir
+		newCmd.Env = slices.Clone(cmd.Env)
+		out, err = newCmd.CombinedOutput()
+		if err != nil && isSSHError(out) {
+			t.Skipf("SSH Connection failed on tape %q again, it's not an authd issue, ignoring it!\n%s\n",
+				td.Name, out)
```

Comment:
Do you have any data on
* how high the percentage of failures is?
* how long these tests take?

I want to have a better idea of how many cycles (and time we have to wait for the tests to pass) we're spending here by retrying before skipping.

Reply:
> how high the percentage of failures is?

Very low, I almost never saw it on CI, but I got it locally at times

> how long these tests take?

Not much, it sometimes it just fails at startup because it can't connect (yet?) to SSH, so retrying just works.... I was unsure if skipping it all the times, but since retrying always worked to me, I decided to take this road.

Reply:
Ok, if it happens rarely and the test fails early in case of this error, wouldn't it make more sense to 1. let the test fail if it fails on retry, so that we don't miss any potential actual issues by skipping the test, and 2. retry more often if we see the test fail on retry?